Buildings/Fluid/Humidifiers/DXDehumidifier.mo

```diff
@@ -0,0 +1,347 @@
+
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToReal booToRea
+    "Convert enable signal from Boolean to Real"
+    annotation (Placement(transformation(extent={{-90,-60},{-70,-40}})));
+
+  Modelica.Blocks.Routing.RealPassThrough QHea if addPowerToMedium
+    "Heat transfer into medium only if required"
+    annotation (Placement(transformation(extent={{-50,20},{-30,40}})));
+
+  Buildings.Controls.OBC.CDL.Reals.Multiply u
+    "Calculate non-zero humidity removal from inlet air only when component is enabled"
+    annotation (Placement(transformation(extent={{-46,-50},{-26,-30}})));
+
+  Buildings.Fluid.Sensors.MassFractionTwoPort senMasFra(
+    redeclare package Medium = Medium,
+    final m_flow_nominal=mAir_flow_nominal)
+    "Inlet air water vapor mass fraction"
+    annotation (Placement(transformation(extent={{-40,-10},{-20,10}})));
+
+  Buildings.Fluid.Sensors.RelativeHumidityTwoPort senRelHum(
+    redeclare package Medium = Medium,
+    final m_flow_nominal=mAir_flow_nominal)
+    "Inlet air relative humidity"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+
+  Buildings.Controls.OBC.CDL.Reals.MultiplyByParameter eneFac(
+    final k=eneFac_nominal/(1000*1000*3600))
+    "Convert energy factor unit"
+    annotation (Placement(transformation(extent={{0,-100},{20,-80}})));
+
+  Modelica.Blocks.Sources.RealExpression eneFacModVal(
+    final y=eneFacMod)
+    "Calculated value of energy factor modifier curve"
+    annotation (Placement(transformation(extent={{-40,-100},{-20,-80}})));
+
+  Buildings.Controls.OBC.CDL.Reals.MultiplyByParameter watRemRat(
+    final k=VWat_flow_nominal)
+    "Calculate water removal rate by multiplying water removal modifier by nominal removal rate"
+    annotation (Placement(transformation(extent={{0,-70},{20,-50}})));
+
+  Buildings.Controls.OBC.CDL.Reals.Divide PDeh
+    "Calculate dehumidification power consumption"
+    annotation (Placement(transformation(extent={{40,-80},{60,-60}})));
+
+protected
+  constant Modelica.Units.SI.SpecificEnthalpy h_fg= Buildings.Utilities.Psychrometrics.Constants.h_fg
+    "Latent heat of water vapor";
+
+  constant Modelica.Units.SI.Density rhoWat=1000
+    "Water density";
+
+  Real watRemMod(
+    min=0,
+    nominal=1,
+    start=1)
+    "Water removal modifier factor as a function of temperature and relative humidity";
+
+  Real eneFacMod(
+    min=0,
+    nominal=1,
+    start=1)
+    "Energy factor modifier factor as a function of temperature and relative humidity";
+
+equation
+    //-------------------------Part-load performance modifiers----------------------------//
+    // Compute the water removal and energy factor modifier fractions, using a biquadratic curve.
+    // Since the regression for capacity can have negative values
+    // (for unreasonable inputs), we constrain its return value to be
+    // non-negative.
+    watRemMod =Buildings.Utilities.Math.Functions.smoothMax(
```

Comment:
In general, do not mix the graphical modeling with the textual code. For this mode, to achieve it, a base class might be needed.

Reply:
@karthikeyad-pnnl 
Please let me know if you need help on addressing the comment.